wiki/C++-Style-Guide.md

```diff
@@ -0,0 +1,814 @@
+
+## Line length
+
+Try to keep lines to about 80 characters in length, and wrap anything over 120 characters.
+Keeping the code all on one line is preferable in most cases, even if you have to rewrite the line into two separate statements to make it short enough. Wrapped lines are hard to read, but so are super-long lines.
+
+## Large vertical breaks
+
+Put three blank lines between function definitions, and before and after the class definition in a .h file.
+Aside from files themselves, these are the largest "units" the code is broken up into. If your code is well-written, the reader should not need to keep looking back and forth between multiple functions, so it does not matter that the extra space will make it so fewer functions fit on the screen at once.
+
+
+
+## Medium vertical breaks
+
+Put two blank lines between sections in a class definition, or to split up sections of code paragraphs in a large function.
+Of course, if a function is that long it might need to be refactored anyways.
+
+## Small vertical breaks
+
+Use single blank lines to group lines of code into "paragraphs," both within class definitions and within functions.
+The rules here are analogous to paragraphs in prose: start with an introductory sentence (e.g. a comment summarizing what follows) and make sure that each "paragraph" is only dealing with one thing.
+
+## Indentation in blank lines
+
+Blank lines should not be indented.
+This is to avoid conflicts between different editors, and to make style checking easier.
+
+## One expression per line
+
+Do not put multiple expression on the same line.
+Except, of course, for the three individual clauses contained in a for loop.
+
+## Parentheses
+
+Do not put a space before or after the parentheses for functions and control statements.
+There is no consensus on this among programmers. But, the mathematical convention is to never put a space, i.e. you never write "sin (x)". Much of our code is mathematical in nature, so it makes sense to follow that convention.
+As part of that convention, also don't put spaces inside the parentheses (like "sin( x )"). The only exceptions from this rule are macro calls in the unit tests:
+```c++
+SCENARIO( "Creating an Account" , "[Account][Creation]" ) {
+    GIVEN( "an account" ) {
+        Account account;
+        WHEN( "money is added" ) {
+            ...
+        }
+    }
+}
+```
+If you are used to putting spaces before the parentheses, rather than trying to adjust your typing you might find it easier to just do a find/replace before committing your code, e.g. replacing "if (" with "if(", etc. You can also grep the svn diff of your changes for "^+.*\ (" to check if you missed anything.
```

Comment:
```suggestion
If you are used to putting spaces before the parentheses, rather than trying to adjust your typing you might find it easier to just do a find/replace before committing your code, e.g. replacing `if (` with `if(`, etc. You can also grep the svn diff of your changes for `^+.*\ (` to check if you missed anything.
```